crates/chia-consensus/src/run_block_generator.rs

```diff
@@ -382,12 +382,14 @@ where
             continue;
         };
         let coinspend = CoinSpend::new(coin, puzzle_program.clone(), solution_program.clone());
```

Comment:
you can probably defer constructing this object until when you need it, in case the puzzle fails